source/vstd/slice.rs

```diff
@@ -10,18 +10,30 @@ pub use super::std_specs::vec::VecAdditionalSpecFns;
 
 verus! {
 
-pub trait SliceAdditionalSpecFns<T> {
+impl<T> View for [T] {
+    type V = Seq<T>;
+
     spec fn view(&self) -> Seq<T>;
+}
+
+impl<T: DeepView> DeepView for [T] {
+    type V = Seq<T::V>;
+
+    open spec fn deep_view(&self) -> Seq<T::V> {
+        let v = self.view();
+        Seq::new(v.len(), |i: int| v[i].deep_view())
+    }
```

Comment:
Just making a quick note here that @y1ca1 had some concerns (during a separate discussion) about this adding more triggers than explicitly reimplementing `deep_view` on each of the vstd data structures individually.

Personally I think that implementing `deep_view` separately on each of (say) `[u8]`, `[u32]`, `[u64]` and so on is not really a good idea (because it doesn't scale and would prevent usage on user-defined types without them manually _also_ implementing deepview on `[Foo]`), and that we should just go with the current approach.

However, @y1ca1, if I've misunderstood what you meant, please do comment here.

Reply:
Thanks for mentioning this. What I meant by "explicitly reimplementing" `deep_view` is to provide separate implementation on each of the methods for the generic slice type`[T]` whose specs are described in `deep_view`. This would arguably save a lot of triggers if one uses `deep_view` ubiquitously, but yeah, definitely less ergonomic and harder to maintain.